src/adapter/src/catalog/apply.rs

```diff
@@ -504,6 +512,236 @@ impl CatalogState {
         );
     }
 
+    #[instrument(level = "debug")]
+    fn apply_system_object_mapping_update(
```

Comment:
Most of this is lifted directly from `open.rs`, except for applying builtin types (`Builtin::Type(typ)` in the match below) is very different.